gateway-spi/src/main/java/org/apache/knox/gateway/topology/Service.java

```diff
@@ -122,32 +122,61 @@ public boolean equals(Object object) {
       return false;
     }
     Service that = (Service) object;
-    String thatName = that.getName();
+    String thatName = that.name;
     if (thatName != null && !(thatName.equals(name))) {
         return false;
     }
-    String thatRole = that.getRole();
+    String thatRole = that.role;
     if (thatRole != null && !thatRole.equals(role)) {
         return false;
     }
-    Version thatVersion = that.getVersion();
+    Version thatVersion = that.version;
     if (thatVersion != null && !(thatVersion.equals(version))) {
         return false;
     }
+
+    // URLs
+    if (urls.size() != that.urls.size()) {
+      return false;
+    }
+
+    for (String url : urls) {
+      if (!that.urls.contains(url)) {
+        return false;
+      }
+    }
+
+    // Params
+    if (params.size() != that.params.size()) {
+      return false;
+    }
+
+    for (Map.Entry<String, String> param : params.entrySet()) {
+      if (!param.getValue().equals(that.params.get(param.getKey()))) {
+        return false;
+      }
+    }
+
     return true;
   }
 
   @Override
   public int hashCode() {
```

Comment:
Adding `HashCodeBuilder` and `EqualsBuilder` here too?

Reply:
+1